Framework/PythonInterface/mantid/utils/dgs/_geometry.py

```diff
@@ -0,0 +1,299 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+from collections import namedtuple
+from collections.abc import Mapping
+import functools
+from typing import Any, Tuple
+import numpy as np
+from mantid.geometry import OrientedLattice
+import mantid.kernel as mk
+from enum import Enum
+import warnings
+warnings.filterwarnings("ignore", category=RuntimeWarning)
+
+
+__all__ = ['qangle', 'ErrorCodes']
+
+
+def namedtuplefy(func):
+    r"""
+    Decorator to transform the return dictionary of a function into
+    a namedtuple
+    Parameters
+    ----------
+    func: Function
+        Function to be decorated
+    name: str
+        Class name for the namedtuple. If None, the name of the function
+        will be used
+    Returns
+    -------
+    Function
+    """
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        res = func(*args, **kwargs)
+        if wrapper.nt is None:
+            if isinstance(res, Mapping) is False:
+                raise ValueError('Cannot namedtuplefy a non-dict')
+            wrapper.nt = namedtuple(func.__name__ + '_nt', res.keys())
+        return wrapper.nt(**res)
+    wrapper.nt = None
+    return wrapper
+
+
+def _get_angle_from_vectors(x: np.array,
```

Comment:
Since this is dead code, i.e. it's not called anywhere, can we get rid of it until we have an implementation for it? These sort of functions get left to rot, especially if a future developer doesn't notice it and implements it elsewhere

Reply:
I will